core/src/main/java/com/linecorp/armeria/client/AttributeMapMaybeWithParent.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright 2019 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.client;
+
+import java.util.Iterator;
+
+import io.netty.util.Attribute;
+import io.netty.util.AttributeKey;
+import io.netty.util.AttributeMap;
+
+/**
+ * Holds {@link Attribute}s which can be accessed via {@link AttributeKey}.
+ */
+// TODO(minwoox) Better name for this?
+public interface AttributeMapMaybeWithParent extends AttributeMap {
```

Comment:
Time to fork `AttributeMap` interface? I guess this breaking change will affect few users.

Reply:
That's a good idea. So the `AttribuiteMap` will be:
```java
public interface AttributeMap {

    <T> Attribute<T> attr(AttributeKey<T> key);
    <T> Attribute<T> ownAttr(AttributeKey<T> key);
    <T> boolean hasAttr(AttributeKey<T> key);
    <T> boolean hasOwnAttr(AttributeKey<T> key);
    Iterator<Attribute<?>> attrs();
    Iterator<Attribute<?>> ownAttrs();
}
```
Is that right?

Reply:
On second thought, how about adding the `own` methods to `DefaultAttributeMap` and `ClientRequestContext` only? No need to fork anything and introduce a new API.

Reply:
`ClientRequestContext` has to call `ServiceRequestContext.defaultAttribuiteMap()`. But `DefaultAttributeMap` is internal API so I can't add it to the public API because it violates the :site:build. That's why I introduce this new interface.